app/experimenter/experiments/tests/test_forms.py

```diff
@@ -1954,3 +1955,39 @@ def test_form_removes_subscribers(self):
 
         experiment = form.save()
         self.assertFalse(self.user in experiment.subscribers.all())
+
+
+class TestNormandyIdForm(MockRequestMixin, TestCase):
+
+    def test_form_not_valid_with_bad_other_ids(self):
+        experiment = ExperimentFactory.create()
+
+        form = NormandyIdForm(
+            self.request,
+            instance=experiment,
+            data={"normandy_id": "4343", "other_normandy_ids": "434, aaa"},
+        )
+
+        self.assertFalse(form.is_valid())
+
+    def test_form_not_valid_when_other_ids_duplicate_main_id(self):
+        experiment = ExperimentFactory.create()
+
+        form = NormandyIdForm(
+            self.request,
+            instance=experiment,
+            data={"normandy_id": "4343", "other_normandy_ids": "4343"},
+        )
+
+        self.assertFalse(form.is_valid())
+
+    def test_form_valid_when_other_ids_formatted_correctly(self):
+        experiment = ExperimentFactory.create()
+
+        form = NormandyIdForm(
+            self.request,
+            instance=experiment,
+            data={"normandy_id": "4343", "other_normandy_ids": "4323, 5671"},
+        )
+
+        self.assertTrue(form.is_valid())
```

Comment:
there were no form tests for the NormandyIdForm... so I made some since I added an elaborate clean method... didn't exhaustively test all possible wonky things that could be passed in to the form... but I did try to break this in lots of ways and I think I caught a good portion of the wonky stuff that could be passed in...

Reply:
Yep fantastic, I was just saying this to Jenny earlier today, test the most obvious failure cases, test the happy case, that's good enough until QA finds arcane eldritch incantations to break it in ways that cause our fragile human minds to reel and roil and the piecing together of dissociated knowledge will open up such terrifying vistas of reality, and of our frightful position therein, that we shall either go mad from the revelation or flee from the light into the peace and safety of a new dark age. Ph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn.